services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -182,7 +184,7 @@ public void run() {
           if (topicState.replicationFactor() == -1) {
             return;
           }
-          if (topicState.someBrokerMissingPartition()) {
+          if (topicState.someBrokerMissingPartition() && !topicAssignmentIsRunning()) {
```

Comment:
Why not wait and do it as soon as possible rather checking in another 10 minutes.